### PR TITLE
Regression in Lua column layouts: forward classes correctly.

### LINKED ISSUE #6863: Matplotlib image does not apepar on margin in pdf
Hi,

I'd like to use Tufte style with Matplotlib images for both HTML and PDF. The following code produce expected HTML, however, in PDF, the image is placed in the main text, not at the margin.

````
---
title: "matplotlib demo"
format:
  html: 
      self-contained: true
      grid: 
          margin-width: 350px
  pdf: default
jupyter: python3
---

Demo

```{python}
#| echo: false
#| column: margin

import matplotlib.pyplot as plt

plot = plt.figure()
plot.set_figwidth(2)
plot.set_figheight(3)

x = [1, 2]
y = [2, 3]

plt.plot(x,y)

```
````

Looking at the Latex-file, I can see that \marginnote is just absent. When I try to enclose the code into another {.column-margin} div:

````
::: {.column-margin}
```{python}

code

```
::: 

````
it still works in HTML, but it makes error in PDF, as it produces Latex `figure` environment inside \marginnote.

1. Is it a bug, or I missed an option?
2. Is it possible to get an output of Matplotlib without placing it into `figure`?
3. Otherwise, I will probably need to export image from the Python code and import it inside `{.column-margin}`.

Many thanks!

### PR DESCRIPTION
**Pull request by cscheid**
Closes #6863.

### CODE REVIEW

File: src/resources/filters/layout/columns-preprocess.lua [hunk 1]
```diff
@@ -64,6 +64,13 @@ function resolveColumnClassesForCodeCell(el)
   local figCaptionClasses = float_caption_classes.fig
   local tblCaptionClasses = float_caption_classes.tbl
 
+  quarto.utils.dump {
```
Comment (dragonstyle) [inline src/resources/filters/layout/columns-preprocess.lua:67]: extraneous println

File: src/resources/filters/layout/columns-preprocess.lua [hunk 2]
```diff
@@ -78,12 +85,14 @@ function resolveColumnClassesForCodeCell(el)
           -- look through the children for any figures or tables
           local forwarded = false
           for j, figOrTableEl in ipairs(childEl.content) do
+            print(figOrTableEl)
```
Comment (dragonstyle) [inline src/resources/filters/layout/columns-preprocess.lua:88]: extraneous println

**Review by dragonstyle — COMMENTED** [review]
🔥

File: src/resources/filters/layout/columns-preprocess.lua (hunk 1 above)
Comment (cscheid) [reply src/resources/filters/layout/columns-preprocess.lua:67]: I lost the game.

**Pull request merged**